### PR TITLE
[RFC][UN-15033] Calendar mobile behavior

### PR DESCRIPTION
**Pull request by niceoutput**
https://uniplaces.atlassian.net/browse/UN-15033

screen shot 2018-11-27 at 16 18 09

### CODE REVIEW

**Review by henrmota — APPROVED** [review]

**Review by mmelo — APPROVED** [review]

File: addon/components/uni-datepicker.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { computed } from '@ember/object';
 import { isEmpty } from '@ember/utils';
 import moment from 'moment';
 import layout from '../templates/components/uni-datepicker';
```
Comment (migueldemoura) [inline addon/components/uni-datepicker.js]: Move this down

File: addon/components/uni-datepicker.js [hunk 2]
```diff
@@ -20,6 +21,8 @@ export default Component.extend({
   disabledDates: [],
   messages: [],
 
+  media: service(),
```
Comment (migueldemoura) [inline addon/components/uni-datepicker.js]: Move this to the top

File: addon/templates/components/uni-datepicker.hbs [hunk 3]
```diff
@@ -3,8 +3,17 @@
   selected=selected
   onCenterChange=(action onCenter value="moment")
   onSelect=(action onSelect value="moment")
+  classNames=(if isFullScreen 'mobile')
```
Comment (migueldemoura) [inline addon/templates/components/uni-datepicker.hbs]:
```suggestion
  classNames=(if isFullScreen "mobile")
```

File: addon/templates/components/uni-datepicker.hbs [hunk 4]
```diff
@@ -3,8 +3,17 @@
...
 as |calendar|}}
 
+  {{yield}}
+
+  {{#if media.isMobile}}
+    <div class="close-mobile-modal" {{action "closeModalMobile"}}>
```
Comment (migueldemoura) [inline addon/templates/components/uni-datepicker.hbs]:
```suggestion
    <div class="close-mobile-modal" {{action "onClose"}}>
```

File: addon/components/uni-datepicker.js [hunk 5]
```diff
@@ -52,6 +56,10 @@ export default Component.extend({
 
     prevMonth() {
       this.get('onCenter')(this.get('center').clone().add(-1, 'month'));
+    },
+
+    closeModalMobile() {
```
Comment (migueldemoura) [inline addon/components/uni-datepicker.js]:
```suggestion
    onClose() {
```

**Review by migueldemoura — CHANGES_REQUESTED** [review]
Just minor comments 👍

File: addon/templates/components/uni-datepicker.hbs (hunk 4 above)
Comment (DaniFrancisco) [reply addon/templates/components/uni-datepicker.hbs]: `close-modal` only, it shouldn't be necessarily attached to the mobile usecase

File: addon/templates/components/uni-datepicker.hbs [hunk 6]
```diff
@@ -3,8 +3,17 @@
   selected=selected
   onCenterChange=(action onCenter value="moment")
   onSelect=(action onSelect value="moment")
+  classNames=(if isFullScreen 'mobile')
 as |calendar|}}
 
+  {{yield}}
+
+  {{#if media.isMobile}}
```
Comment (DaniFrancisco) [inline addon/templates/components/uni-datepicker.hbs]: `isFullscreen`

**Review by DaniFrancisco — CHANGES_REQUESTED** [review]
Tests missing

File: tests/integration/components/uni-datepicker-test.js [hunk 7]
```diff
@@ -49,3 +49,10 @@ test('when there is no messages the message container does not render', function
 
   assert.dom('.message').doesNotExist();
 });
+
+test('test to see if the modal went fullscreen', function(assert) {
+
```
Comment (DaniFrancisco) [inline tests/integration/components/uni-datepicker-test.js:54]: rm

**Review by DaniFrancisco — APPROVED** [review]

File: tests/integration/components/uni-datepicker-input-test.js [hunk 8]
```diff
@@ -10,5 +11,19 @@ test('it renders', function(assert) {
 
   this.render(hbs`{{uni-datepicker-input}}`);
 
-  assert.equal(this.$().text().trim(), '');
+  assert.dom('.uni-datepicker-input').exists();
+});
+
+test('test to see if the date picker modal toggle between clicks', async function(assert) {
```
Comment (migueldemoura) [inline tests/integration/components/uni-datepicker-input-test.js]:
```suggestion
test('it toggles date picker component existence when clicking input', async function(assert) {
```

File: tests/integration/components/uni-datepicker-test.js [hunk 9]
```diff
@@ -49,3 +49,10 @@ test('when there is no messages the message container does not render', function
 
   assert.dom('.message').doesNotExist();
 });
+
+test('test to see if the modal went fullscreen', function(assert) {
```
Comment (migueldemoura) [inline tests/integration/components/uni-datepicker-test.js]: Not sure this test should be for this component.

**Review by migueldemoura — CHANGES_REQUESTED** [review]
Just noticed that you're not using the new test syntax, which is mandatory. I'd change it as you'll have to do it when merging the FB anyway.

**Review by migueldemoura — APPROVED** [review]

**Pull request merged**